tiledb/sm/crypto/CMakeLists.txt

```diff
@@ -37,8 +37,8 @@ commence(object_library tiledb_crypto)
         this_target_link_libraries(bcrypt)
     else()
         find_package(OpenSSL REQUIRED)
-	#this_target_link_libraries(OpenSSL::Crypto)
-	target_link_libraries(tiledb_crypto PRIVATE OpenSSL::Crypto)
+        #this_target_link_libraries(OpenSSL::Crypto)
+        target_link_libraries(tiledb_crypto PRIVATE OpenSSL::Crypto)
```

Comment:
What's the resolution on this? It hasn't been addressed from last time.

Reply:
The difference is that `this_target_link_libraries` links with `PUBLIC` visibility, which is undesirable in this case, because we use OpenSSL as an internal implementation detail. Changing it right now does not break anything, but it will have undesirable effects once the `tiledb_crypto` object library is incorporated to the main build (and the `OpenSSL::Crypto` target would be a requirement even for shared libraries).

I am going to remove the commented line, and add a comment on why we can't use `this_target_link_libraries`.

Reply:
Done.